electrumsv/network.py

```diff
@@ -142,6 +142,11 @@ async def _synchronise_headers_for_server_tip(self, server_state: HeaderServerSt
         """
         assert app_state.headers is not None
         tip_headers = await get_chain_tips_async(server_state, self.aiohttp_session)
+        # The `new_server_connection_event` is set here immediately following the first successful
```

Comment:
This comment says "This is now set here and previously it wasn't". I do not know why it is moved. What was the observed problem?

Reply:
There was no observed problem, only theoretical. `wait_until_header_server_is_ready_async` waits on this `new_server_connection_event` and to me it is wrong to have it unblock before the server has been successfully connected to.

Previously it was launching the async task and immediately setting this event regardless of success or failure - doesn't seem correct.

Reply:
What failure makes that a problem?

Reply:
`start_reference_server_connection_async` will then potentially proceed against a server that is down.
If you're not happy with it I can put it back but to me it just seems like common sense and gives the expected behaviour for a function called `wait_until_header_server_is_ready_async`.

If the server hasn't even been connected to yet - how can you say it is "ready"?

Reply:
I don't think that's how `wait_until_header_server_is_ready_async` works.

Reply:
This will never be hit unless the header server task hasn't even been started.
```
        while server_key not in self.connected_header_server_states:
            await self.new_server_connection_event.wait()
```

This will wait for the headers to sync
```
        server_state = self.connected_header_server_states[server_key]
        await server_state.connection_event.wait()
```

Reply:
Okay. Yes that makes sense. Will revert that change and force put the latest commit now.